Support printing labels from Media

Sigmonia opened the pull request:
#### Rationale
https://github.com/LabKey/biologics/issues/1533
Spec: https://docs.google.com/document/d/1yXBuQdaenDtmoTiIRWDqfB3sOytVjsp8SV8HPLVbiKE/edit#

#### Related Pull Requests
* [Biologics](https://github.com/LabKey/biologics/pull/1559)
* [Sample Manager](https://github.com/LabKey/sampleManagement/pull/1200)

#### Changes
* Move the Print modal to the SamplesTabbedGrid

Sigmonia committed:
switch to use SamplesTabbedGridPanel
Move PrintLabelsModal

Sigmonia committed:
post merge revamp

Sigmonia committed:
Add flag to show/hide label printing option even if user is allowed to print labels

Sigmonia committed:
Merge remote-tracking branch 'origin/develop' into fb_mediaBartender

Sigmonia committed:
v2.213.1-fb-mediaBartender.0

cnathe commented on packages/components/package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@labkey/components",
-  "version": "2.213.1",
+  "version": "2.213.1-fb-mediaBartender.0",
```
don't forget the update to `components.md` release notes

cnathe commented on packages/components/src/internal/components/samples/SamplesTabbedGridPanel.tsx:
```diff
@@ -256,6 +260,29 @@ export const SamplesTabbedGridPanel: FC<Props> = memo(props => {
+    }, []);
+
+    const afterPrint = useCallback((numSamples: number, numLabels: number): void => {
+        setPrintDialogModel(undefined);
```
looks like `setShowPrintModal` is never set back to `false` here. I think that state variable can actually just go away and just use the `printDialogModel` state var instead. So it would show the modal if the `printDialogModel` is defined and then setting it back to undefined would hide the modal.

cnathe commented on packages/components/src/internal/components/samples/SamplesTabbedGridPanel.tsx:
```diff
@@ -304,7 +331,7 @@ export const SamplesTabbedGridPanel: FC<Props> = memo(props => {
                     ButtonsComponentRight={SampleTabbedGridButtonsRight}
-                    supportedExportTypes={canPrintLabels ? EXPORT_TYPES_WITH_LABEL : undefined}
+                    supportedExportTypes={showLabelOption && canPrintLabels ? EXPORT_TYPES_WITH_LABEL : undefined}
                     onExport={canPrintLabels ? onLabelExport : undefined}
```
should this `onExport` also be conditional on the `showLabelOption` prop as well?

cnathe commented on packages/components/src/internal/components/search/FindSamplesByIdsPageBase.tsx:
```diff
@@ -130,19 +103,6 @@ export const FindSamplesByIdsTabbedGridPanelImpl: FC<FindSamplesByIdsTabProps> =
                 }}
                 user={user}
```
this usage should now set `showLabelOption` true right? otherwise, the Find By Ids page now no longer shows the print labels in the export menu.

Sigmonia committed:
merge from develop

Sigmonia committed:
v2.214.1-fb-mediaBartender.1

Sigmonia committed:
Add draft release notes
remove redundant flag

Sigmonia committed:
make property optional with default

Sigmonia committed:
v2.214.1-fb-mediaBartender.2

Sigmonia committed:
adjust display casing to match model

Sigmonia committed:
Add an initialContext prop to support jest tests for LabelPrintingContext
Add LabelPrintingContext to test Context stack
Add null user check (some test situations don't provide a user)

Sigmonia committed:
v2.214.1-fb-mediaBartender.3

cnathe commented on packages/components/src/internal/components/labels/utils.ts:
```diff
@@ -1,5 +1,5 @@
 
 export function userCanPrintLabels(user: User): boolean {
-    return !user.isGuest;
+    return user && !user.isGuest;
```
should add a quick jest test case to SamplesTabbedGridPanel for this with showLabelOption true but user as guest

cnathe commented on packages/components/src/internal/components/samples/SamplesTabbedGridPanel.spec.tsx:
```diff
@@ -201,15 +201,15 @@ describe('SamplesTabbedGridPanel', () => {
         wrapper.unmount();
     });
 
     test('canPrintLabels true', () => {
```
update names of tests to be `showLabelOption false` and `showLabelOption true`?

Sigmonia replied:
added a couple additional tests to cover both

cnathe commented on packages/components/src/internal/testHelpers.tsx:
```diff
@@ -118,17 +122,20 @@ export const AppContextTestProvider: FC<AppContextTestProviderProps> = props =>
     notificationContext?: Partial<NotificationsContextState>,
-    options?: MountRendererProps
+    options?: MountRendererProps,
+    printLabelsContext?: Partial<LabelPrintingProviderProps>,
```
should the `printLabelsContext` param go before `options`? or are you worries about breaking existing usages?

Sigmonia replied:
the latter

Sigmonia committed:
Add print context 'canPrintLabels' flag and showLabelOption flag tests

Sigmonia committed:
Adjust initial print context to use supplied user

Sigmonia committed:
v2.214.1-fb-mediaBartender.4

Sigmonia committed:
v2.215.0-fb-mediaBartender.5

Sigmonia committed:
finalize version and release notes.
